snowfall/decoding/second_pass.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c)  2021  Xiaomi Corporation (authors: Fangjun Kuang)
+
+# This file implements the ideas proposed by Daniel Povey.
+#
+# See https://github.com/k2-fsa/snowfall/issues/232 for more details
+#
+
+import k2
+
+
+class Nbest(object):
```

Comment:
@danpovey 

This file implements the `Nbest` class proposed in
https://github.com/k2-fsa/snowfall/issues/232#issue-941386780

Reply:
That's great!  Yes it looks like what I had in mind.
I assume you would separate it from this PR though?  Or maybe even submit it to k2?  Since there's a lot going on here.

Reply:
Will move it to k2.